chessgame/src/common/pieces/King.ts

```diff
@@ -0,0 +1,26 @@
+import Piece from "./_Interface";
+import tails from "../constant/tails";
+
+class King extends Piece {
+    type: string = "King";
+
+    canMove(column: string, row: number): boolean {
+        const ccn: number = tails[this.column];     //current column number
+        const ncn: number = tails[column];          //new column number
+
+        if(Math.abs(ccn - ncn) == 1 && row == this.row ){
```

Comment:
To mozna w sumie wszystko jednym warunkiem załatwic, ze columnDifference i rowDifference sa <=1 i ze nie sa rowne zero obie na raz